pkg/resource/deploy/plan_executor.go

```diff
@@ -260,6 +262,33 @@ func (pe *planExecutor) Execute(callerCtx context.Context, opts Options, preview
 		return res
 	}
 
+	// If the step generator and step executor were both successful, then we send all the resources
+	// observed to be analyzed. Otherwise, this step is skipped.
+	if res == nil && !pe.stepExec.Errored() {
+		resourcesSeen := pe.stepGen.urns
+		resources := make([]plugin.AnalyzerResource, 0, len(resourcesSeen))
+		for _, v := range resourcesSeen {
+			resources = append(resources, plugin.AnalyzerResource{
+				Type: v.Type,
+				// Unlike Analyze, AnalyzeStack is called on the final outputs of each resource,
+				// to verify the final stack is in a compliant state.
+				Properties: v.Outputs,
+			})
+		}
+
+		analyzers := pe.stepGen.plan.ctx.Host.ListAnalyzers()
+		for _, analyzer := range analyzers {
+			diagnostics, aErr := analyzer.AnalyzeStack(resources)
+			if aErr != nil {
+				return result.FromError(aErr)
+			}
+			for _, d := range diagnostics {
+				pe.stepGen.hasPolicyViolations = pe.stepGen.hasPolicyViolations || (d.EnforcementLevel == apitype.Mandatory)
+				pe.stepGen.opts.Events.OnPolicyViolation("" /* don't associate with any particular URN */, d)
```

Comment:
We should consider passing the URN along with each resource, and providing an API to report the URN associated with a violation (I have a proposal for the API on the TypeScript/JavaScript side to do this).

That way, if you have a rule about a particular resource that requires looking at other (related) resources in the stack (as part of the "analyze stack" check), you can associate the violation more precisely to the resource where the issue actually is rather than a vague "there's a policy violation somewhere in the stack" error.

We should also consider passing along more information with each resource:
  - parent URN
  - dependency URNs
  - resources options